etc/scripts/release.hs

```diff
@@ -89,17 +92,32 @@ main =
                 gProjectRoot = "" -- Set to real value velow.
                 gBuildArgs = []
                 gCertificateName = Nothing
-                global0 = foldl (flip id) Global{..} flags
+                globalDefault = Global{..}
+
+            -- modify our default value of type Global with env vars and command line args
+            envOptions <- envOptionsIO
+            let globalWithEnv = envOptions globalDefault
+                globalWithArgsEnv = foldl (flip id) globalWithEnv flags
+
             -- Need to get paths after options since the '--arch' argument can effect them.
-            projectRoot' <- getStackPath global0 "project-root"
-            let global = global0
+            projectRoot' <- getStackPath globalWithArgsEnv "project-root"
+            let global = globalWithArgsEnv
                     { gProjectRoot = projectRoot' }
             return $ Just $ rules global args
   where
     getStackPath global path = do
       out <- readProcess stackProgName (stackArgs global ++ ["path", "--" ++ path]) ""
       return $ trim $ fromMaybe out $ stripPrefix (path ++ ":") out
 
+-- | Env Vars to set some of the command-line options found below.
+envOptionsIO :: IO (Global -> Global)
+envOptionsIO = do
```

Comment:
Why did you take a totally different approach than is used for handling the GITHUB_AUTH_TOKEN variable?  I would have thought you could just copy that approach since it's basically doing the same thing (only difference is that there's a default).  E.g. in `main` have `gGpgKey <- fromMaybe "0x575159689BEFB442" <$> lookupEnv gpgKeyEnvVar` instead of `let gGpgKey = "0x575159689BEFB442"`.  Much shorter code, and also more efficient than iterating through all the environment variables.

While in there, the default key should be moved to a `defaultGpgKey` constant instead of being a magic value here (not sure why I didn't do that originally).